### PR TITLE
parser,config: Correct handling of display width deprecation except for zerofill and tinyint(1)

### LINKED ISSUE #46650: Missing display width for integer types with deprecate-integer-display-length
## Bug Report

In MySQL 8.0.19 and newer (including MySQL 8.1.0) integer display width is hidden by default.

Except when:
- The type is `tinyint` and the display width is `1` as connectors use this to indicate booleans.
- When `ZEROFILL` is used.

TiDB with `deprecate-integer-display-length` doesn't do this, which might cause issues with some connectors.

From https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-19.html

> Display width specification for integer data types was deprecated in MySQL 8.0.17, and now statements that include data type definitions in their output no longer show the display width for integer types, with these exceptions:
> 
> The type is TINYINT(1). MySQL Connectors make the assumption that TINYINT(1) columns originated as BOOLEAN columns; this exception enables them to continue to make that assumption.
> 
> The type includes the ZEROFILL attribute.

### 1. Minimal reproduce step (Required)

```sql
CREATE TABLE t (
  id INT PRIMARY KEY,
  c1 BOOL,
  c2 INT(10) ZEROFILL
);
SHOW CREATE TABLE t\G
```

### 2. What did you expect to see? (Required)

MySQL 8.1.0
```
sql> SHOW CREATE TABLE t\G
*************************** 1. row ***************************
       Table: t
Create Table: CREATE TABLE `t` (
  `id` int NOT NULL,
  `c1` tinyint(1) DEFAULT NULL,
  `c2` int(10) unsigned zerofill DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
1 row in set (0.0037 sec)
```
### 3. What did you see instead (Required)

```
sql> SHOW CREATE TABLE t\G
*************************** 1. row ***************************
       Table: t
Create Table: CREATE TABLE `t` (
  `id` int NOT NULL,
  `c1` tinyint DEFAULT NULL,
  `c2` int unsigned zerofill DEFAULT NULL,
  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
1 row in set (0.0004 sec)
```

### 4. What is your TiDB version? (Required)

Mon 4 Sep build from master at 587eacf6ba28c57e61974dba75627238d0e883a7

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #46650 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Display width deprecation now excludes `tinyint(1)` and types with `ZEROFILL`. This improves compatibility with MySQL Connectors.
```

### CODE REVIEW

**Comment by dveeden** [comment]
I removed the change for the default value of `DeprecateIntegerDisplayWidth` and only fix the handling of `tinyint(1)` and `zerofilll`. Then we can change the default later.

File: executor/test/showtest/show_test.go [hunk 1]
```diff
@@ -1647,12 +1647,22 @@ func TestShowCreateTableWithIntegerDisplayLengthWarnings(t *testing.T) {
...
+		"  `a` tinyint(1) DEFAULT NULL,\n" +
 		"  `b` smallint DEFAULT NULL,\n" +
 		"  `c` mediumint DEFAULT NULL,\n" +
 		"  `d` int DEFAULT NULL,\n" +
 		"  `e` bigint DEFAULT NULL\n" +
 		") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
+
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(id int primary key, c1 bool, c2 int(10) zerofill)")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1064 You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use [parser:1681]Integer display width is deprecated and will be removed in a future release."))
```
Comment (mjonss) [inline executor/test/showtest/show_test.go:1659]: Should we have the same warnings as MySQL (below 8.0.34)?
```
mysql> create table t(id int primary key, c1 bool, c2 int(10) zerofill);
Query OK, 0 rows affected, 2 warnings (0,03 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                                                                                                   |
+---------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Warning | 1681 | The ZEROFILL attribute is deprecated and will be removed in a future release. Use the LPAD function to zero-pad numbers, or store the formatted numbers in a CHAR column. |
| Warning | 1681 | Integer display width is deprecated and will be removed in a future release.                                                                                              |
+---------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
2 rows in set (0,00 sec)
```

**Review by mjonss — APPROVED** [review]
LGTM, just a question if we should also have a warning for ZEROFILL being deprecated? (should we create an issue for that or just accept that we are not raising a warning, like MySQL does?)

File: executor/test/showtest/show_test.go (hunk 1 above)
Comment (dveeden) [reply executor/test/showtest/show_test.go:1659]: I think that would be good, but also outside of the scope of this PR.

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**